app/models/environment.rb

```diff
@@ -79,7 +79,11 @@ def auto_deploy_enabled?
 
   # The default git ref to deploy when none is provided for this environment.
   def default_ref
-    super.presence || self.class.default_ref
+    client_default_branch || super.presence || self.class.default_ref
```

Comment:
Does this mean the github default branch takes precedence? Want to make sure we're not overriding things in a way that would break functionality, although I guess since we discovered it's not actually configurable maybe it's fine?

Reply:
Yes, the default branch would take precedence. And yea, it falls back to a hardcoded master and we can't set this in the config as you point out.